lib/components/MonthPicker/MonthPicker.tsx

```diff
@@ -0,0 +1,252 @@
+      year: value && value.year ? value.year : undefined,
+      month: parseInt(nativeValue, 10) || undefined,
+    },
+    year: {
+      month: value && value.month ? value.month : undefined,
+      year: parseInt(nativeValue, 10) || undefined,
+    },
+    native: stringToCustomValue(nativeValue),
+  }[fieldType]);
+
+const handleChange = <Element extends HTMLSelectElement | HTMLInputElement>(
+  { onChange, value }: Pick<MonthPickerProps, 'onChange' | 'value'>,
+  fieldType: FieldType,
+) => (event: ChangeEvent<Element>) => {
+  if (typeof onChange === 'function') {
+    onChange(nativeToCustomValue(event.target.value, value, fieldType), event);
+  }
+};
+
+const handleBlur = <Element extends HTMLSelectElement | HTMLInputElement>(
+  { onBlur, value }: Pick<MonthPickerProps, 'onBlur' | 'value'>,
+  fieldType: FieldType,
+) => (event: FocusEvent<Element>) => {
+  if (typeof onBlur === 'function') {
+    const isRelatedTargetExternal =
+      event.relatedTarget !== monthRef.current &&
+      event.relatedTarget !== yearRef.current;
+
+    if (event.relatedTarget === null || isRelatedTargetExternal) {
+      onBlur(nativeToCustomValue(event.target.value, value, fieldType), event);
+    }
+  }
+};
+
+const handleFocus = <Element extends HTMLSelectElement | HTMLInputElement>({
+  onFocus,
+}: Pick<MonthPickerProps, 'onFocus'>) => (event: FocusEvent<Element>) => {
+  if (typeof onFocus === 'function') {
+    const isRelatedTargetExternal =
+      event.relatedTarget !== monthRef.current &&
+      event.relatedTarget !== yearRef.current;
+
+    if (event.relatedTarget === null || isRelatedTargetExternal) {
+      onFocus(event);
+    }
+  }
+};
+
+export const MonthPicker = ({
+  id,
+  name,
+  value,
+  label,
+  onChange,
+  onBlur,
+  onFocus,
+  tone,
+  disabled,
+  minYear = currYear - 100,
+  maxYear = currYear,
+  ascendingYears = false,
+  ...restProps
+}: MonthPickerProps) => {
+  const styles = useStyles(styleRefs);
+  const currentValue = {
+    month: value && value.month ? value.month : undefined,
+    year: value && value.year ? value.year : undefined,
+  };
+
+  const field = (
+    <Field
+      id={`${id}-month`}
+      name={`${name}-month`}
+      tone={tone}
+      disabled={disabled}
+      label={
+        <Box component={renderNativeInput ? 'span' : 'legend'}>
+          {label}
+          <Box component="span" display="none">
+            Month
+          </Box>
+        </Box>
+      }
+      {...restProps}
+      autoComplete={undefined}
+      secondaryMessage={null}
+    >
+      {({
+        backgroundColor,
+        boxShadow,
+        width,
+        paddingLeft,
+        paddingRight,
+        borderRadius,
+        className,
+        ...nonVisualFieldProps
+      }) =>
+        renderNativeInput ? (
+          <Box
+            component="input"
+            type="month"
+            value={customValueToString(currentValue)}
+            onChange={onChange && handleChange({ onChange, value }, 'native')}
```

Comment:
I thought this was a bug because it looked like it was firing the change handler immediately during render. I feel like these would be better named as something like `makeChangeHandler`. What do you think?